yt/data_objects/time_series.py

```diff
@@ -56,16 +57,12 @@ def _eval(params, ds):
                 dict(eval = _eval, _params = tuple()))
     return cls
 
-def get_filenames_from_glob_pattern(filenames):
-    file_list = glob.glob(filenames)
-    if len(file_list) == 0:
-        data_dir = ytcfg.get("yt", "test_data_dir")
-        pattern = os.path.join(data_dir, filenames)
-        td_filenames = glob.glob(pattern)
-        if len(td_filenames) > 0:
-            file_list = td_filenames
-        else:
-            raise YTOutputNotIdentified(filenames, {})
+def get_filenames_from_glob_pattern(pattern):
+    epattern = os.path.expanduser(pattern)
+    data_dir = ytcfg.get("yt", "test_data_dir")
+    file_list = glob.glob(epattern) or glob.glob(os.path.join(data_dir, epattern))
```

Comment:
> This function changes the behavior of the part after the "or", right? It would be os.path.join(data_dir, pattern) to match the original behavior.

oh you're right, this is actually broken as it stands !

Reply:
> I also see why this reduces the code to one line, but I do think this is less clear. Maybe an inline comment saying that if glob.glob(epattern) is None then we switch to the join part?

you got it !

Reply:
This function changes the behavior of the part after the "or", right? It would be os.path.join(data_dir, pattern) to match the original behavior.
oh you're right, this is actually broken as it stands !
on second thought, I'm not so sure this was broken...
Let me assume the user has data under $test_data_dir/mydata/ and try to discover it with

ts = DatasetSeries("~/mydata/*")
No matter where $test_data_dir points to, prepending ~ with it will never match the file tree, so using os.path.expanduser shouldn't change anything.
Am I missing something here ?

Reply:
you're not missing anything. If this update makes it more inclusive, that's fine with me! I just noticed that it was different behavior compared to the original function. 🙂